source/common/http/path_utility.h

```diff
@@ -24,5 +30,24 @@ class PathUtil {
   static absl::string_view removeQueryAndFragment(const absl::string_view path);
 };
 
+class PathTransformer {
+public:
+  PathTransformer(envoy::type::http::v3::PathTransformation const& path_transformation);
+
+  // Take a string_view as argument and return an optional string.
+  // The optional will be null if the transformation fail.
+  absl::optional<std::string> transform(const absl::string_view original_path) const;
```

Comment:
Nit: could you mention that this will execute all configured transformations, or something along those lines?

Reply:
Thanks, will add comments.